from minio import Minio
from minio.error import ResponseError

def main(args):
    minioClient = Minio(endpoint='{{ service_ip }}',
                    access_key='minio-access-key',
                    secret_key='minio-secret-key',
                    secure=False)

    # # List Buckets
    # buckets = minioClient.list_buckets()
    # for bucket in buckets:
    #     print(bucket.name, bucket.creation_date)

    # # Download a full object.
    # try:
    #     data = minioClient.get_object('openwhisk', 'sample.png')
    #     with open('my-minio-file.png', 'wb') as file_data:
    #         for d in data.stream(32*1024):
    #             file_data.write(d)
    # except ResponseError as err:
    #     print(err)

    try:
        get = minioClient.fget_object('openwhisk', 'sample.png', '/tmp/sample.png')
        put = minioClient.fput_object('openwhisk', 'new-sample.png', '/tmp/sample.png', content_type='image/png')
        return {"get": str(get), "put": str(put)}
    except Exception as e:
        return {'error': str(e)} 
    
